vllm/v1/worker/gpu_model_runner.py

```diff
@@ -655,7 +655,10 @@ def _prepare_inputs(
 
         # Fill unused with -1. Needed for reshape_and_cache
         self.seq_lens[num_reqs:].fill_(0)
-        self.query_start_loc[num_reqs + 1:].fill_(-1)
+        # Note: pad query_start_loc to be non-decreasing, as kernels
+        # like FlashAttention requires that
+        self.query_start_loc[num_reqs + 1:].fill_(
+            self.query_start_loc_cpu[num_reqs].item())
```

Comment:
I think `.item()` will cause 2-way GPU-CPU transfer for a single value, any way we could avoid that?

Reply:
It's a cpu tensor here.

Reply:
Oh my bad - that seems fine then